qwen/qwen-vl/model/model.py

```diff
@@ -0,0 +1,57 @@
+import base64
+import os
+import tempfile
+from io import BytesIO
+
+from PIL import Image
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+BASE64_PREAMBLE = "data:image/png;base64,"
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self.model = None
+        self.tokenizer = None
+
+    def load(self):
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            "Qwen/Qwen-VL", trust_remote_code=True
+        )
+        self.model = AutoModelForCausalLM.from_pretrained(
+            "Qwen/Qwen-VL", device_map="cuda", trust_remote_code=True
+        ).eval()
+
+    def b64_to_pil(self, b64_str):
+        return Image.open(
+            BytesIO(base64.b64decode(b64_str.replace(BASE64_PREAMBLE, "")))
+        )
+
+    def predict(self, request: dict):
+        image = request.pop("image")
+        prompt = request.pop("prompt")
+
+        created_temp_file = False
+        if not image.startswith("http") or not image.startswith("https"):
```

Comment:
is the image valid otherwise? Will the model work on b64 or do you need to handle the else statement?